src/js/jsx/search/SearchBar.jsx

```diff
@@ -213,82 +216,131 @@ define(function (require, exports, module) {
         /**
          * Find options to show in the Datalist drop down
          *
-         * @param {Immutable.List.<object>} options Full list of potential options
          * @param {string} searchTerm Term to filter by
+         * @param {string} autofillID ID of option that is currently being suggested
+         * @param {bool} truncate Whether or not to restrict number of options
          * @return {Immutable.List.<object>}
          */
-        _filterSearch: function (options, searchTerm) {
-            // Keep track of how many options shown so far in a given category
-            var count = 0;
+        _filterSearch: function (searchTerm, autofillID, truncate) {
+            // Use count to keep track of how many options shown so far in a given category
+            var count = 0,
+                optionGroups,
+                categories = this._getFilterOptions();
+
+            if (categories.size > 0 && this.state.groupedOptions) {
+                optionGroups = this.state.groupedOptions.unshift(categories);
+            } else {
+                return Immutable.List();
+            }
 
-            return options && options.filter(function (option) {
-                if (option.hidden) {
-                    return false;
-                }
+            var filteredOptionGroups = optionGroups.map(function (options) {
+                return options && options.filter(function (option) {
+                    option.priority = 1; // Add to priority to indicate less important
```

Comment:
It looks like you're mutating the shared options from the search store in this function by setting their priority. Is that right? If so, that feels a little risky, and it would be nicer to find a way to do this without mutating shared state.